src/main/scala/org/ergoplatform/local/MempoolAuditor.scala

```diff
@@ -71,19 +72,21 @@ class MempoolAuditor(nodeViewHolderRef: ActorRef,
       nodeViewHolderRef ! GetNodeViewChanges(history = false, state = true, mempool = true, vault = false)
 
     case ChangedMempool(mp: ErgoMemPoolReader) =>
-      stateReaderOpt.fold[Any](poolReaderOpt = Some(mp))(initiateCleanup(_, mp))
+      poolReaderOpt = Some(mp)
```

Comment:
This was a bug?

Reply:
I guess so

Reply:
Then maybe add tests to check the internal state is changed consistently.

Reply:
Made new "rebroadcast transactions correctly" test in MempoolAuditorSpec, where a MempoolAuditor instance is receiving ChangedState and ChangedMempool signals, and then we're checking that it is doing cleanup and rebroadcasting transactions indeed.